jormungandr/src/fragment/selection.rs

```diff
@@ -64,106 +143,58 @@ impl FragmentSelectionAlgorithm for OldestFirst {
         soft_deadline_future: futures::channel::oneshot::Receiver<()>,
         hard_deadline_future: futures::channel::oneshot::Receiver<()>,
     ) -> (Contents, ApplyBlockLedger) {
-        use futures::future::{select, Either};
-
         let date: BlockDate = ledger.block_date().into();
-        let mut current_total_size = 0;
+        let mut space_left = ledger_params.block_content_max_size;
         let mut contents_builder = ContentsBuilder::new();
         let mut return_to_pool = Vec::new();
-
         let soft_deadline_future = soft_deadline_future.shared();
         let hard_deadline_future = hard_deadline_future.shared();
-
         while let Some(fragment) = pool.remove_oldest() {
             let id = fragment.id();
-            let fragment_raw = fragment.to_raw(); // TODO: replace everything to FragmentRaw in the node
-            let fragment_size = fragment_raw.size_bytes_plus_size() as u32;
-
-            let span = span!(Level::TRACE, "fragment_selection_algorithm", kind="older_first", hash=%id.to_string());
-            let _enter = span.enter();
-            if fragment_size > ledger_params.block_content_max_size {
-                let reason = format!(
-                    "fragment size {} exceeds maximum block content size {}",
-                    fragment_size, ledger_params.block_content_max_size
-                );
-                tracing::debug!("{}", reason);
-                logs.modify(id, FragmentStatus::Rejected { reason }, date);
-                continue;
-            }
-
-            let total_size = current_total_size + fragment_size;
-
-            if total_size > ledger_params.block_content_max_size {
-                // return a fragment to the pool later if does not fit the contents size limit
-                return_to_pool.push(fragment);
-                continue;
-            }
-
-            tracing::debug!("applying fragment in simulation");
-
-            let fragment1 = fragment.clone();
-            let ledger1 = ledger.clone();
-            let fragment_future =
-                tokio::task::spawn_blocking(move || ledger1.apply_fragment(&fragment1));
-
-            let result = match select(fragment_future, soft_deadline_future.clone()).await {
-                Either::Left((join_result, _)) => join_result.unwrap(),
-                Either::Right((_, fragment_future)) => {
-                    if current_total_size > 0 {
-                        tracing::debug!(
-                            "aborting processing of the current fragment to satisfy the soft deadline"
-                        );
-                        return_to_pool.push(fragment);
-                        break;
+            let span = debug_span!("fragment", hash=%id.to_string()[..7]);
```

Comment:
We already have a `Display` impl for hash, it was just to avoid putting the whole hash in the logs span, which clutters the plain text logs.
I don't think we should treat the plain-text version as B-series, but I agree it's not ideal for other backends. If it's not possible to format depending on the backend, printing the full hash will be fine I guess.